### PR TITLE
Group booking confirmation page

### PR DESCRIPTION
**Pull request by silksil**

### CODE REVIEW

File: src/containers/booking-confirmation/BookingRewards.tsx [hunk 1]
```diff
@@ -1,41 +1,47 @@
...
   );
 };
 
 export const BookingRewards = () => {
-  const [params] = useSearchParams();
-  const offerId = params.get('offerId');
-  const { data, isLoading, claimReward, error } = useRewards(offerId);
+  const { bookingMode, bookingInfo } = useCheckout();
+  const isGroupMode = bookingMode === 'group';
+  const offerId = !isGroupMode ? getOfferId(bookingInfo?.offers) : undefined;
```
Comment (dave-ok) [inline src/containers/booking-confirmation/BookingRewards.tsx:34]: I am curious why you didn't just return the offerId from the `useCheckout` hook

File: src/containers/booking-confirmation/BookingDetails.tsx [hunk 2]
```diff
@@ -2,11 +2,18 @@ import { useSearchParams } from 'react-router-dom';
 import { Box, Typography } from '@mui/material';
 import { ExternalLink } from 'src/components/ExternalLink';
 import { useAppState } from 'src/store';
+import { useCheckout } from '../../hooks/useCheckout/useCheckout';
```
Comment (dave-ok) [inline src/containers/booking-confirmation/BookingDetails.tsx:5]: You may need to modify this to `index` as you did with the previous PRs

File: src/containers/booking-confirmation/BookingRewards.tsx (hunk 1 above)
Comment (silksil) [reply src/containers/booking-confirmation/BookingRewards.tsx:34]: Write now the `bookingInfo` state is used for a normal booking and a group booking. There is only ` offer` key within `bookInfo` not a `offerId` key, as the `offerId` is part of the `offer`. Thus, in order to get the `offerId` you pass the `offer` which then extracts the offerId.

Maybe i can make it more  intuitive so I am open to reaftor it.

**Review by dave-ok — APPROVED** [review]
Looks great!

File: src/containers/booking-confirmation/BookingRewards.tsx [hunk 3]
```diff
@@ -1,41 +1,47 @@
...
+  const queryId = isGroupMode ? bookingInfo?.requestId : offerId;
```
Comment (bobetbat) [inline src/containers/booking-confirmation/BookingRewards.tsx:35]: can be simplified, `queryId = isGroupMode ? bookingInfo?.requestId : getOfferId(bookingInfo?.offers);`

File: src/hooks/useRewards.ts [hunk 4]
```diff
@@ -12,20 +12,24 @@ interface ClaimRewardResponse {
...
+
+  const { data } = await axios.get<RewardOption[]>(url).catch((_) => {
+    throw new Error('Could not retrieve rewards');
+  });
 
   return data;
 };
 
-const postClaimReward = async ({ id, rewardType }: MutationProps) => {
+const postClaimReward = async ({ id, rewardType }: MutationProps, isGroupBooking) => {
```
Comment (bobetbat) [inline src/hooks/useRewards.ts:27]: missed type

**Pull request merged**